test/cpp/end2end/ssl_credentials_test.cc

```diff
@@ -85,51 +131,252 @@ class SslCredentialsTest : public ::testing::Test {
+    std::shared_ptr<Channel> channel =
+        grpc::CreateCustomChannel(server_addr_, creds, channel_args);
+
+    auto stub = grpc::testing::EchoTestService::NewStub(channel);
+    grpc::testing::EchoRequest request;
+    grpc::testing::EchoResponse response;
+    request.set_message(kMessage);
+    ClientContext context;
+    context.set_deadline(grpc_timeout_seconds_to_deadline(/*time_s=*/10));
+    grpc::Status result = stub->Echo(&context, request, &response);
+    if (!result.ok()) {
+      return absl::Status(static_cast<absl::StatusCode>(result.error_code()),
+                          result.error_message());
+    }
+    EXPECT_EQ(response.message(), kMessage);
+    return context.auth_context();
+  }
+
+  void ExpectResumedSession(std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ("true", ToString(properties[0]));
+  }
+
+  void ExpectNonResumedSession(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_SSL_SESSION_REUSED_PROPERTY);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ("false", ToString(properties[0]));
+  }
+
+  void ExpectOk(const absl::Status& status) {
+    EXPECT_EQ(status.code(), absl::StatusCode::kOk);
+    EXPECT_EQ(status.message(), "");
+  }
+
+  void ExpectTransportSecurityTypeIsSsl(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties = auth_context->FindPropertyValues(
+        GRPC_TRANSPORT_SECURITY_TYPE_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), GRPC_SSL_TRANSPORT_SECURITY_TYPE);
+  }
+
+  void ExpectTransportSecurityLevelIsPrivacyAndIntegrity(
+      std::shared_ptr<const AuthContext> auth_context) {
+    std::vector<grpc::string_ref> properties = auth_context->FindPropertyValues(
+        GRPC_TRANSPORT_SECURITY_LEVEL_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), "TSI_PRIVACY_AND_INTEGRITY");
+  }
+
+  void ExpectPeerCommonNameEquals(
+      std::shared_ptr<const AuthContext> auth_context,
+      absl::string_view common_name) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_CN_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), common_name);
+  }
+
+  void ExpectPeerSubjectEquals(std::shared_ptr<const AuthContext> auth_context,
+                               absl::string_view subject) {
+    std::vector<grpc::string_ref> properties =
+        auth_context->FindPropertyValues(GRPC_X509_SUBJECT_PROPERTY_NAME);
+    ASSERT_EQ(properties.size(), 1u);
+    EXPECT_EQ(ToString(properties[0]), subject);
+  }
+
+  void ExpectPeerSubjectAltNameEquals(
```

Comment:
Instead of having this method, you can do the following in the individual tests:

```
absl::string_view subject_alt_name =
    GetAuthContextProperty(*auth_context, GRPC_X509_SAN_PROPERTY_NAME);
EXPECT_THAT(
    absl::StrSplit(subject_alt_name, ","),
    ::testing::UnorderedElementsAre(
        "*.test.google.fr", "waterzooi.test.google.be",
        "*.test.youtube.com", "192.168.1.3"));
```

Same thing for the other SAN methods below.

Reply:
Good call. I adjusted this slightly and created a separate `GetAuthContextPropertyAsSet` function, because the `GetAuthContextProperty` function sometimes returns strings (e.g. the common name) that contain commas.